pipeline/testing/tests_selftest/test_dedup.py

```
@@ -0,0 +1,55 @@
+        a = read("crash-a.txt")
+        b = read("crash-b.txt")
+        report = dedup.dedup_texts([a, b], labels=["a", "b"], strict_lines=True)
+        self.assertEqual(report["unique_fingerprints"], 2)
+
+    def test_no_traceback_fallback(self):
+        fp1, _ = dedup.fingerprint("not a traceback at all")
+        fp2, _ = dedup.fingerprint("not a traceback at all   ")
+        self.assertEqual(fp1, fp2)
+
+    def test_cli_exit_code(self):
+        proc = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)